pkg/controller/jobframework/reconciler.go

```diff
@@ -239,6 +241,13 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 			err := r.startJob(ctx, job, object, wl)
 			if err != nil {
```

Comment:
This means that the Job never starts, but also never gets a Failed condition, meaning that it's stuck suspended until the user manually deletes it.
But that's no worse that the current behavior, so +1 on the change.